src/components/Modals/Snaps/SnapConfirm.tsx

```diff
@@ -11,27 +11,76 @@ import {
   ModalHeader,
   UnorderedList,
 } from '@chakra-ui/react'
-import React, { useCallback, useState } from 'react'
+import { isEvmChainId } from '@shapeshiftoss/chain-adapters'
+import { knownChainIds } from 'constants/chains'
+import React, { useCallback, useEffect, useState } from 'react'
 import { useTranslate } from 'react-polyglot'
 import { enableShapeShiftSnap } from 'utils/snaps'
 import { CircularProgress } from 'components/CircularProgress/CircularProgress'
 import { RawText } from 'components/Text'
+import { useIsSnapInstalled } from 'hooks/useIsSnapInstalled/useIsSnapInstalled'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { walletSupportsChain } from 'hooks/useWalletSupportsChain/useWalletSupportsChain'
 import { getMixPanel } from 'lib/mixpanel/mixPanelSingleton'
 import { MixPanelEvent } from 'lib/mixpanel/types'
+import { useAppDispatch } from 'state/store'
+
+import { fetchAccountForChainId } from './helpers'
 
 type SnapConfirmProps = {
   onClose: () => void
 }
 
 export const SnapConfirm: React.FC<SnapConfirmProps> = ({ onClose }) => {
   const [isInstalling, setIsInstalling] = useState(false)
+  const [isInstallationRequestComplete, setIsInstallationRequestComplete] = useState(false)
   const [hasAgreed, setHasAgreed] = useState(false)
   const [hasPinkySworeSeedPhraseIsBackedUp, setHasPinkySworeSeedPhraseIsBackedUp] = useState(false)
   const translate = useTranslate()
-  const handleAddSnap = useCallback(() => {
+  const { state: walletState } = useWallet()
+  const dispatch = useAppDispatch()
+  const isSnapInstalled = useIsSnapInstalled()
+
+  const discoverNonEvmAccounts = useCallback(async () => {
+    const { wallet, deviceId: walletDeviceId } = walletState
+
+    if (!wallet) return
+
+    const nonEvmChainIds = knownChainIds.filter(
+      chainId =>
+        !isEvmChainId(chainId) &&
+        walletSupportsChain({
+          isSnapInstalled: true,
+          chainId,
+          wallet,
+          checkConnectedAccountIds: false,
+        }),
+    )
+
+    // fetch account 0 for each non-evm chain
+    await Promise.all(
+      nonEvmChainIds.map(async chainId => {
+        await fetchAccountForChainId(chainId, wallet, walletDeviceId, dispatch, true, 0)
+      }),
+    )
+  }, [dispatch, walletState])
```

Comment:
This sounds off too - this should be an `<AppContext />` concern? This isn't automagical anymore as far as code is concerned